Modules/Sources/WordPressUI/Views/DataView/DataViewPaginatedForEach.swift

```diff
@@ -0,0 +1,37 @@
+import SwiftUI
+
+/// A view that displays paginated data using ForEach with automatic loading triggers.
+public struct DataViewPaginatedForEach<Response: DataViewPaginatedResponseProtocol, Content: View>: View {
+    @ObservedObject private var response: Response
+    private let content: (Response.Element) -> Content
+
+    /// Creates a paginated ForEach view.
+    ///
+    /// - Parameters:
+    ///   - response: The paginated response handler that manages the data.
+    ///   - content: A view builder that creates the content for each item.
+    public init(
+        response: Response,
```

Comment:
If I'm not mistaken, the query is built into the `Response` instance. When you change the query (like sorting or filtering by a different property), you create a new `Response` instance and thus use a different `DataViewPaginatedForEach` view. If that's the case, the list would start from the first page after changing the query, since previously loaded data is lost (no matter how far the user has scrolled), which may not be a good UX?

It's not something we need to solve now, but I thought it might be something we should consider further down the road.